### PR TITLE
replicating routes-services PR

### PR DESCRIPTION
**Pull request by maksimlikharev**
producing single PR for routes/services

### CODE REVIEW

**Comment by DonMartin76** [comment]
Please resolve merge conflicts. Thanks!

**Comment by maksimlikharev** [comment]
21: updateStep21_v1_0_0i,
<<<<<<< HEAD
    22: updateStep21_MultiRoutes
=======
    22: updateStep22_v1_0_0m
>>>>>>> 268b2a60b3a34543e989cfdd016512eba3c6dcd8

I already moved version from 21 to 22, I cannot move it anymore, routes has to be merged and updateStep22_v1_0_0m should go to version 23

File: src/env/src/config-updater.js [hunk 1]
```diff
@@ -259,6 +262,39 @@ function updateStep21_v1_0_0i(targetConfig, sourceConfig, configKey) {
     saveGlobals(targetConfig, targetGlobals);
 }
 
+function updateStep21_MultiRoutes(targetConfig, sourceConfig, configKey) {
```
Comment (DonMartin76) [inline src/env/src/config-updater.js]: It looks a little disturbing that the function name is "21", but it's actually step 22.

**Comment by DonMartin76** [comment]
There are some tests failing, presumably because you also update the API section of the auth servers (which is good), but you didn't adapt the tests for it.

**Comment by DonMartin76** [comment]
```
  1) /auth-server
       /<auth-server-id>
         should only return very few properties when retrieving as non-admin:
     Uncaught AssertionError: Missing config.api property: expected undefined to be truthy
      at Function.assert.isOk (node_modules/chai/lib/chai/interface/assert.js:85:32)
      at Request._callback (test/12_authServerTests.js:75:24)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at endReadableNT (_stream_readable.js:1145:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  2) /auth-server
       /<auth-server-id>
         should return all properties when retrieving as admin:
     Uncaught AssertionError: Missing config.api.uris property: expected undefined to be truthy
      at Function.assert.isOk (node_modules/chai/lib/chai/interface/assert.js:85:32)
      at Request._callback (test/12_authServerTests.js:95:24)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at endReadableNT (_stream_readable.js:1145:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)

  3) /auditlog
       auditlog?embed=1
         must be possible to retrieve auditlog as an admin and filter by role and not find any row by 'Admin' role:
     Uncaught AssertionError: expected '1' to equal 0
      at Request._callback (test/19_auditlogTest.js:210:28)
      at Request.self.callback (node_modules/request/request.js:185:22)
      at Request.<anonymous> (node_modules/request/request.js:1161:10)
      at IncomingMessage.<anonymous> (node_modules/request/request.js:1083:12)
      at endReadableNT (_stream_readable.js:1145:12)
      at process._tickCallback (internal/process/next_tick.js:63:19)
```

File: src/api/routes/apis.js [hunk 2]
```diff
@@ -514,10 +514,21 @@ apis.getSwagger = function (app, res, loggedInUserId, apiId) {
         const globalSettings = utils.loadGlobals(app);
         const configJson = loadApiConfig(app, apiId);
 
-        if (!configJson || !configJson.api || !configJson.api.uris || !configJson.api.uris.length) {
+        const requestPaths = [];
+
+        if ( configJson && configJson.api && configJson.api.routes ) {
```
Comment (DonMartin76) [inline src/api/routes/apis.js]: Please stick to the coding style around you (spaces, parenthesis etc.).

**Comment by maksimlikharev** [comment]
config update is a big mess rightnow, it's getting more and more difficult to deal with it gracefully

**Comment by maksimlikharev** [comment]
ok, fixed conflicts, fixed unit test as well + other notes

**Comment by DonMartin76** [comment]
Regarding the config update: I agree. Open for ideas. But it's not so simple.

**Comment by DonMartin76** [comment]
I am pulling this in now; it may still be that I will be requesting changes.

**Pull request merged**